pkg/api/validation/validation.go

```diff
@@ -2754,10 +2759,22 @@ func ValidateResourceRequirements(requirements *api.ResourceRequirements, fldPat
 		if api.IsStandardResourceName(string(resourceName)) {
 			allErrs = append(allErrs, validateBasicResource(quantity, fldPath.Key(string(resourceName)))...)
 		}
+		if resourceName == api.ResourceMemory && !IsResourceRequirementsMemoryValid(quantity) {
+			allErrs = append(allErrs, field.Invalid(fldPath, quantity.String(), "memory request must not be configured in millibytes"))
+		}
 	}
 	return allErrs
 }
 
+// validateResourceRequirementsMemory ensures that memory is not configured in milli value
+func IsResourceRequirementsMemoryValid(quantity resource.Quantity) bool {
```

Comment:
There should be a test case that demonstrates the failure scenario that is covered by this change.